packages/medusa/src/services/payment-provider.ts

```diff
@@ -368,11 +369,14 @@ export default class PaymentProviderService extends TransactionBaseService {
     }
   }
 
-  async createPayment(
-    cart: Cart & { payment_session: PaymentSession }
-  ): Promise<Payment> {
+  async createPayment(data: {
+    id: string
+    total: number
+    region: Region
+    payment_session: PaymentSession
```

Comment:
suggestion: wdyt of this instead? 
```suggestion
    cartId: string
    amount: number
    currency_code: string
    payment_session: PaymentSession
```